app/models/tube.rb

```diff
@@ -90,12 +90,28 @@ def self.create_with_barcode!(*args, &block)
     validate_barcode(barcode, prefix) if barcode.present?
     barcode ||= AssetBarcode.new_barcode
 
+    concentration = attributes.delete(:concentration)
+    volume = attributes.delete(:volume)
+    parent = attributes.delete(:parent)
+    aliquot_attributes = attributes.delete(:aliquot_attributes)
+
     # remove this so it's not passed in on creation, and set it explicitly afterwards
     # this is to control the order of barcode addition so that it gets set as the 'primary' barcode
     foreign_barcode = attributes.delete(:foreign_barcode)
 
     tube = create!(attributes.merge(sanger_barcode: { prefix: prefix, number: barcode }), &block)
 
+    receptacle = tube.receptacle
+    receptacle.qc_results.build(key: 'molarity', value: concentration, units: 'nM')
+    receptacle.qc_results.build(key: 'volume', value: volume, units: 'ul')
+    target = receptacle.transfer_requests_as_target.build(
+      asset: parent.receptacle,
```

Comment:
I don't think you can move this into the main Tube class, most tubes will not have QC information or be created from a parent.
Shall we zoom and discuss?

Reply:
Yeah I just wanted to see whether if it works outside the block. Wanted to locate where the issue actually was. It did confirm that it was the block in `create!` that causes all these havoc. 

I'll arrange a zoom. I'm planning to go through some Rails upgrade documentations to see if they have made changes to the create function and to see what they are. Is it alright if I arrange a Zoom after that?